pandas/core/nanops.py

```diff
@@ -372,7 +372,7 @@ def _wrap_results(result, dtype: np.dtype, fill_value=None):
             if isna(result):
                 result = np.datetime64("NaT", "ns")
             else:
-                result = np.int64(result).view("datetime64[ns]")
+                result = np.int64(result).view(dtype)
```

Comment:
can you do the nat case a couple lines up too just to be on the safe side (like we do for td64 below)